halo/evmstaking2/keeper/keeper_internal_test.go

```diff
@@ -88,18 +88,112 @@ func TestInsertAndDeleteEVMEvents(t *testing.T) {
 	}
 }
 
-func TestHappyPathDelivery(t *testing.T) {
+func TestDeliveryWithBrokenServer(t *testing.T) {
```

Comment:
Need godocs for all tests

Reply:
we don't really do this for any tests in the code base. So not sure we can start a new convention simply in the middle of a PR review?

Reply:
I did some quick googling and it does not look like this convention is a thing, in general...

I think comments inside the test doing the "handholding" and explaining the steps are more useful, aren't they? If the godoc would contain all these details instead, you'd still need to do mental work by mapping the docs to the actual code. But if we remove the details, then the test name is already explaining the high level idea, doesn't it?

Reply:
> we don't really do this for any tests in the code base. So not sure we can start a new convention simply in the middle of a PR review?
I think this is fine starting good standards as we go along, especially in small PR's. 

> I did some quick googling and it does not look like this convention is a thing, in general...
> 
> I think comments inside the test doing the "handholding" and explaining the steps are more useful, aren't they? If the godoc would contain all these details instead, you'd still need to do mental work by mapping the docs to the actual code. But if we remove the details, then the test name is already explaining the high level idea, doesn't it?

Brief godoc that concisely describes the test and expected behavior is very useful. Example [geth](https://github.com/ethereum/go-ethereum/blob/master/core/state_processor_test.go#L43). That way without any context a dev can see what the test does, just looking at the code is not enough. I dont think there is any cost to adding godocs for all tests (atleast the more involved ones) and all exported funcs.